src/lava/lib/optimization/solvers/generic/solver.py

```diff
@@ -0,0 +1,227 @@
+CPUS = [CPU, "CPU"]
+NEUROCORES = [Loihi2NeuroCore, NeuroCore, "Loihi2"]
+
+
+def solve(problem: OptimizationProblem,
+          timeout: int,
+          target_cost: int = None,
+          backend: BACKENDS = Loihi2NeuroCore) -> \
+        npt.ArrayLike:
+    """Create solver from problem spec and run until target_cost or timeout.
+
+    Parameters
+    ----------
+    problem: Optimization problem to be solved.
+    timeout: Maximum number of iterations (timesteps) to be run. If set to
+    -1 then the solver will run continuously in non-blocking mode until a
+     solution is found.
+    target_cost: A cost value provided by the user as a target for the
+    solution to be found by the solver, when a solution with such cost is
+    found and read, execution ends.
+    backend: Specifies the backend where the main solver network will be
+    deployed.
+
+    Returns
+    ----------
+    solution: candidate solution to the input optimization problem.
+    """
+    solver = OptimizationSolver(problem)
+    solution = solver.solve(timeout=timeout, target_cost=target_cost,
+                            backend=backend)
+    return solution
+
+
+class OptimizationSolver:
+    """Generic solver for constrained optimization problems defined by
+    variables, cost and constraints.
+
+    The problem should behave according to the OptimizationProblem's
+    interface so that the Lava solver can be built correctly.
+
+    A Lava OptimizationSolverProcess and a Lava OptimizationSolverModel will
+    be created from the problem specification. The dynamics of such process
+    implements the algorithms that search a solution to the problem and
+    reports it to the user.
+
+    Parameters
+    ----------
+    problem: Optimization problem to be solved.
+    run_cfg: Run configuration for the OptimizationSolverProcess.0
+
+    """
+
+    def __init__(self,
+                 problem: OptimizationProblem,
+                 run_cfg=None):
+        self.problem = problem
+        self._run_cfg = run_cfg
+        self._process_builder = SolverProcessBuilder()
+        self.solver_process = None
+        self.solver_model = None
+
+    @property
+    def run_cfg(self):
+        """Run configuration for process model selection."""
+        return self._run_cfg
+
+    @run_cfg.setter
+    def run_cfg(self, value):
+        self._run_cfg = value
+
+    def solve(self,
+              timeout: int,
+              target_cost: int = 0,
+              backend: BACKENDS = CPU,
+              hyperparameters: ty.Dict[
+                  str, ty.Union[int, npt.ArrayLike]] = None) \
+            -> npt.ArrayLike:
+        """Create solver from problem spec and run until target_cost or timeout.
+
+        Parameters
+        ----------
+        timeout: Maximum number of iterations (timesteps) to be run. If set to
+        -1 then the solver will run continuously in non-blocking mode until a
+         solution is found.

```

Comment:
It is well conceivable that the solver will never find a solution, isn't it? Thus, I would provide a maximum number of time steps. Bad style to enable an infinite loop by normal usage of a method.